surefire-providers/surefire-junit-platform/src/test/java/org/apache/maven/surefire/junitplatform/JUnitPlatformProviderTest.java

```diff
@@ -766,6 +768,72 @@ public void tagExpressionsAreSupportedForIncludeTagsContainingAmpersand()
         assertEquals( 2, provider.getFilters().length );
     }
 
+    @Test
+    public void onlyJunitIncludeEngineIsDeclared()
+    {
+        Map<String, String> properties = singletonMap( JUNIT_INCLUDE_ENGINE_PROP, "engine-one, engine-two" );
+
+        ProviderParameters providerParameters = providerParametersMock( TestClass1.class );
+        when( providerParameters.getProviderProperties() ).thenReturn( properties );
+
+        JUnitPlatformProvider provider = new JUnitPlatformProvider( providerParameters );
+
+        assertEquals( 1, provider.getFilters().length );
```

Comment:
Can you examine the content of the filter with the assertion statement?
Assert that the filter contains two inclusive filters "engine-one, engine-two".
Is it possibel, and also in the next tests?

Reply:
Hmm I checked and to verify ids I need to do probably some conversion of junit classes etc.
There is no direct method to get those ids values but only method to apply if TestEngine is included or excluded by filters.
This logic is more inside Junit itself so maybe enough will be to check only if expected filter is instance of EngineFilter?

Reply:
`instanceOf` would be fine too.
Dont use assertEquals, you have more modern style, see `import static org.assertj.core.api.Assertions.assertThat`
Example:
`assertThat( provider.getFilters() ).hasSize( 1 )`
`assertThat( provider.getFilters() ).element( 0 ).isInstanceOf( TestEngine.class )`

Reply:
Hmm there was no method: ".element()" so I used something else :)